public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -78,13 +137,27 @@ export class DashboardGrid extends PureComponent<Props> {
         panelPos.isDraggable = panel.collapsed;
       }
 
-      layout.push(panelPos);
+      if (!panelFilter) {
+        layout.push(panelPos);
+      } else {
+        if (panelFilter.test(panel.title)) {
+          panelPos.isResizable = false;
+          panelPos.isDraggable = false;
+          panelPos.x = (count % 2) * GRID_COLUMN_COUNT;
+          panelPos.y = Math.floor(count / 2);
+          layout.push(panelPos);
+          count++;
```

Comment:
how do you restore the position when the filtering is removed?

Reply:
`panelPos` is created every time the layout is recomputed which would happen on the `panelFilter` changing